msprime/demography.py

```diff
@@ -1982,12 +1982,16 @@ def get_growth_rate(epoch):
         # Add pulses in reverse order, so that pulses with the same time
         # correspond to the correct backwards-time mass migration ordering.
         for pulse in reversed(events.pop("pulses")):
-            demography.add_mass_migration(
-                time=pulse.time,
-                source=pulse.dest,
-                dest=pulse.source,
-                proportion=pulse.proportion,
-            )
+            total_ancestry_moved = 0
+            for prop, source in zip(pulse.proportions, pulse.sources):
+                demography.add_mass_migration(
+                    time=pulse.time,
+                    source=pulse.dest,
+                    dest=source,
+                    proportion=prop / (1 - total_ancestry_moved),
+                )
+                total_ancestry_moved += prop
+            assert total_ancestry_moved == sum(pulse.proportions)
```

Comment:
We need to either delete the asertion or change to ``math.isclose(total_ancestry_moved, sum(pulse.proportions))``

It seems like something we should just do some unit tests for though. Should be easy to make a few examples and test that the output has the required properties?

Reply:
If we're happy that the unit tests are covering this, we don't need the assertion. And, we should be happy that the unit tests are covering this :smile:

Reply:
I think you wrote a couple of utility functions for this pattern @jeromekelleher. `_sequential_to_proportions()` and `_proportions_to_sequential()`.

Reply:
Ah yes, better to use those functions then.

Reply:
Yes, that assert was leftover from testing and I forgot to remove it! I'll also update to use those functions as well - I didn't realize they were already in there